### PR TITLE
Configure GitHub Actions to cancel workflow runs on commits

### PR DESCRIPTION
**Pull request by Ao-senXiong**

### CODE REVIEW

**Comment by Ao-senXiong** [comment]
I don't want the old workflow still runs while new commits kick in, like the following screenshot
image

**Review by wmdietl — COMMENTED** [review]
Does this cancel previous runs on the `master` branch? For PRs, cancelling seems good. For `master`, no commit should cancel earlier runs.

**Comment by Ao-senXiong** [comment]
> Does this cancel previous runs on the `master` branch? For PRs, cancelling seems good. For `master`, no commit should cancel earlier runs.

Updated the settings!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,6 +6,12 @@ on:
   pull_request:
     branches: [ "master" ]
 
+# Configure GitHub Actions cancel in progress workflow to avoid redundant runs in pull requests.
+# See: https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs
+concurrency:
+  group: ${{ github.event_name == 'pull_request' && github.head_ref || 'master-group' }}
```
Comment (wmdietl) [inline .github/workflows/ci.yml:12]: How does one parse this expression? Can you add parenthesis to make the grouping clear? In particular the `master-group` in the logical expression is confusing to me...

Comment (Ao-senXiong) [reply .github/workflows/ci.yml]: Now I am using conditional expression in yml for excluding master branch from canceling running CI. 
See https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs#example-only-cancel-in-progress-jobs-on-specific-branches

**Review by wmdietl — APPROVED** [review]
Thanks! The newest version is much easier to read.

**Comment by wmdietl** [comment]
@Ao-senXiong Much easier to read unfortunately doesn't mean it works correctly :-(
If you look at https://github.com/eisop/checker-framework/commits/master/ it looks like CI runs for earlier commits were cancelled by the later commits.
Can you look at what happened?

**Comment by Ao-senXiong** [comment]
> @Ao-senXiong Much easier to read unfortunately doesn't mean it works correctly :-(
> 
> If you look at https://github.com/eisop/checker-framework/commits/master/ it looks like CI runs for earlier commits were cancelled by the later commits.
> 
> Can you look at what happened?

Sure, let take a look!

**Pull request merged**